### PR TITLE
Removes specialist gear ID locks and reworks the spec skill for non-specialists.

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

**Comment by DominikPanic** [comment]
Something like this? @Rohesie

**Comment by Rohesie** [comment]
What kind of locks are the guns under now?

**Comment by DominikPanic** [comment]
No locks since you told me to also remove `WY_RESTRICTED`

**Comment by Rohesie** [comment]
I mean the spec guns. Those are to remain locked until we rework the spec gun skills similarly to the smartgun skill, applying penalties for not having it.

**Comment by DominikPanic** [comment]
I reworked the skill now.

**Comment by Rohesie** [comment]
But it gives no penalties? Check how the smartgun works regarding the skill, and apply similar penalties for the scout rifle and sniper gun. The grenade launcher, pyro flamer and rocket launcher could do a little `do_after` before every shot for the untrained, since accuracy and scatter work a little differently.

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 1]
```diff
@@ -125,40 +124,46 @@ DEFINES in setup.dm, referenced here.
...
 	return ..()
 
+
 /obj/item/weapon/gun/attack_hand(mob/user)
 	var/obj/item/weapon/gun/in_hand = user.get_inactive_hand()
 	if( in_hand == src && (flags_item & TWOHANDED) )
 		unload(user)//It has to be held if it's a two hander.
-	else ..()
+	else 
+		. = ..()
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]:
```suggestion
		return ..()
```

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 2]
```diff
@@ -125,40 +124,46 @@ DEFINES in setup.dm, referenced here.
...
+
 
 /obj/item/weapon/gun/throw_at(atom/target, range, speed, thrower)
 	if( harness_check(thrower) )
 		to_chat(usr, "<span class='warning'>\The [src] clanks on the ground.</span>")
-	else ..()
+	else 
+		. = ..()
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]:
```suggestion
		return ..()
```

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 3]
```diff
@@ -125,40 +124,46 @@ DEFINES in setup.dm, referenced here.
...
 Note: pickup and dropped on weapons must have both the ..() to update zoom AND twohanded,
 As sniper rifles have both and weapon mods can change them as well. ..() deals with zoom only.
 */
 /obj/item/weapon/gun/dropped(mob/user)
-	..()
+	. = ..()
 
 	stop_aim()
-	if (user && user.client)
+	if(user && user.client)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]:
```suggestion
	if(user?.client)
```

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 4]
```diff
@@ -852,12 +847,15 @@ and you're good to go.
 								added_delay = max(fire_delay - 3*user.mind.cm_skills.heavy_weapons, 6)
 						if(GUN_SKILL_SMARTGUN)
 							if(user.mind.cm_skills.smartgun < 0)
-								added_delay += 2*user.mind.cm_skills.smartgun
+								added_delay += -2*user.mind.cm_skills.smartgun
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]:
```suggestion
								added_delay -= 2*user.mind.cm_skills.smartgun
```

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 5]
```diff
@@ -852,12 +847,15 @@ and you're good to go.
...
+						if(GUN_SKILL_SPEC)
+							if(user.mind.cm_skills.spec_weapons < 0)
+								added_delay += -2*user.mind.cm_skills.spec_weapons
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]:
```suggestion
								added_delay -= 2*user.mind.cm_skills.spec_weapons
```

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -357,12 +361,20 @@
 	damage_mult = config.base_hit_damage_mult
 
 
+/obj/item/weapon/gun/launcher/m92/wield(mob/living/user)
+	. = ..()
+	if(user.mind?.cm_skills && user.mind.cm_skills.spec_weapons < 0)
+		do_after(user, 10, TRUE, 5, BUSY_ICON_HOSTILE)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Are you sure this works as intended? This seems to have no effect. It should be as it is in the after-attack.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 7]
```diff
@@ -424,7 +446,9 @@
 		F.updateicon()
 		playsound(F.loc, fire_sound, 50, 1)
 		sleep(10)
-		if(F && F.loc) F.prime()
+		if(F && F.loc)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
		if(F?.loc) 
```

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 8]
```diff
@@ -534,7 +565,8 @@
...
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
		if(F?.loc) 
```

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 9]
```diff
@@ -598,25 +644,33 @@
...
 	return ready_in_chamber()
 
+
 //No such thing
 /obj/item/weapon/gun/launcher/rocket/reload_into_chamber(mob/user)
-	return 1
+	return TRUE
+
 
 /obj/item/weapon/gun/launcher/rocket/delete_bullet(obj/item/projectile/projectile_to_fire, refund = 0)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
/obj/item/weapon/gun/launcher/rocket/delete_bullet(obj/item/projectile/projectile_to_fire, refund = FALSE)
```

**Pull request merged**